replica/compaction_group.hh

```diff
@@ -227,6 +227,11 @@ public:
         return _async_gate;
     }
 
+    // Closes storage group without stopping its compaction groups that might be referenced elsewhere.
+    future<> close() noexcept {
```

Comment:
The name is misleading.
Should be renamed to `shallow_close` to be more visible.